### PR TITLE
fix(iOS): set parent view controller for detached view controllers

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-28209
https://jira.appcelerator.org/browse/TIMOB-27865

Notes-
 1. We need to set parent view controller, where we are adding  ViewController.view on any other controller's view as subview.
 2. In titanium, UiNavigationController, UITabBarController and AVPlayerViewController are added  as subview. So parent is set for these controllers. Parent will be the controller on which these are added as subview.
3.  While testing, be sure to check for UI in orientation change , any random crash. You should never get log- "Presenting view controllers on detached view controllers is discouraged".
4. In Case of Media VideoPlayer test , when entering in full screen and exiting from full screen you should not get log messages like -
"Warning: <AVPlayerViewController: 0x7fc445038e00> is trying to enter full screen, but is not in its view's window's view controller hierarchy. This results in undefined behavior."
4. Test Cases - https://gist.github.com/vijaysingh-axway/e7d652f91d46c72e368af3273c922417

### CODE REVIEW

**Comment by hansemannn** [comment]
Thank you Vijay! I hope this can be tested very extensively to prevent random crashes or memory leaks. Quite afraid of this kind of major change.

**Comment by vijaysingh-axway** [comment]
> Thank you Vijay! I hope this can be tested very extensively to prevent random crashes or memory leaks. Quite afraid of this kind of major change.

@hansemannn Yeah. It will be tested properly. It would be great if you can test it on your app as well.

**Review by janvennemann — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiUITabGroupProxy.m [hunk 1]
```diff
@@ -147,16 +147,34 @@ - (void)windowWillOpen
...
 }
 
 - (void)windowWillClose
 {
   TiUITabGroup *tabGroup = (TiUITabGroup *)self.view;
   if (tabGroup != nil) {
+    // Remove UITabBarController from parent view  controller
+    UITabBarController *tabController = [(TiUITabGroup *)[self view] tabController];
+    [tabController willMoveToParentViewController:nil];
+    [tabController.view removeFromSuperview];
```
[the remark this answers is not in the document]
Comment (vijaysingh-axway) [reply iphone/Classes/TiUITabGroupProxy.m:173]: Actual addSubview is [here](https://github.com/appcelerator/titanium_mobile/pull/12233/files#diff-30207cc237420adbde6370f0966b5ace65298e900041ae900d71769d050ab57dL160) where view is added on container controller's view. So I have added container view management of adding after window is opened. And removing is where controller.view is removed.

**Review by janvennemann — APPROVED** [review]

**Comment by ssekhri** [comment]
The changes look good and the tests mentioned in the ticket and in this PR work fine.
However as this is a major change would run some more regression tests and memory tests as part of FR.

**Comment by ssekhri** [comment]
FR Passed.
Verified on:
Mac OS: 10.15.4
SDK: 9.3.0.v20201105121820
Appc CLI: 8.1.1
JDK: 11.0.6
Node: 12.16.1
Studio: 6.0.0.202005141803
Xcode: 12.3

**Comment by m1ga** [comment]
@sgtcoolguy  you might want to close this issue: https://github.com/appcelerator/titanium_mobile/issues/11651 too if it is fixed

**Pull request merged**